src/http/client.cc

```diff
@@ -390,7 +390,14 @@ future<> client::do_make_request(connection& con, request& req, reply_handler& h
             });
         }
 
-        return handle(rep, con.in(rep)).finally([reply = std::move(reply)] {});
+        return handle(rep, con.in(rep)).finally([reply = std::move(reply), &con] {
```

Comment:
Need a test that fails before this patch and passes after it

Reply:
working on it